### ISSUE TITLE
Error in Embroider enabled apps using `@tagName` with `<Menu>`

### ISSUE THREAD

**Issue by roomman** [GavinJoyce/ember-headlessui#105]
I hit up against [this Embroider bug](https://github.com/embroider-build/embroider/issues/815) today, while trying to add some `LinkTo` components as menu items, using `@tagName`.

I see `Missing component: link-to in...` in the console.

I just updated to the latest Embroider modules and this is still an unresolved issue. I thought I'd add something here so we can keep track. 👍🏻

**Comment by roomman** [comment]
Hey @alexlafroscia, sorry for the delay in coming back on this. 

Minimal reproduction can be [found here](https://github.com/roomman/ember-headlessui-linkto-bug). I've added individual commits, as follows: 

- [The first](https://github.com/roomman/ember-headlessui-linkto-bug/commit/d513c69209848abe54251833bfe4346e7f68f499) simply installs this addon and renders a menu with one item, which is a `link-to`. When the user clicks they transition to the route. 
- [The second](https://github.com/roomman/ember-headlessui-linkto-bug/commit/1a6f47c4f2d90b50cb1026aeec7c778e7d5acb21) represents the most basic setup for enabling Embroider, with no options. The menu behaves as expected.
- [The third](https://github.com/roomman/ember-headlessui-linkto-bug/commit/457fa1fc18089870ea0b1c8bf9f87e53ffcc9771) adds all Embroider options. Crucially (I stepped through, adding each of the options in turn), the `staticComponents` option breaks the build with the following error:
```
Build Error (PackagerRunner) in templates/application.hbs

Module Error (from ../../../../../../../../Users/simon/Sites/ember-headlessui-linkto-bug/node_modules/thread-loader/dist/cjs.js):
Missing component: link-to in templates/application.hbs
```
I'll keep an eye on [the existing bug report](https://github.com/embroider-build/embroider/issues/815) and close this once it's resolved and the build succeeds.

**Comment by bertdeblock** [comment]
This was fixed in v0.48.0 of `@embroider/compat`:
https://github.com/embroider-build/embroider/blob/master/CHANGELOG.md#v0480-2021-12-07

I checked out the reproduction and bumped the embroider versions to latest and everything works fine now.

**Comment by roomman** [comment]
Awesome, thanks @bertdeblock, I'll close this issue.
